### PR TITLE
MetaSearch: add logging

### PR DESCRIPTION
**Pull request by tomkralidis**
## Description

This PR adds basic logging to the MetaSearch panel to help with debugging.

### CODE REVIEW

File: python/plugins/MetaSearch/util.py [hunk 1]
```diff
@@ -167,3 +166,10 @@ def clean_ows_url(url):
         url.setQuery(query_string)
 
     return url.toString()
+
+
+def log_message(message, level):
+    """helper function to emit logging messages"""
+
+    message_translated = QCoreApplication.translate('MetaSearch', message)
+    LOGGER.logMessage(message_translated, 'MetaSearch', getattr(Qgis, level))
```
Comment (GeoSander) [inline python/plugins/MetaSearch/util.py:175]:
```suggestion
    LOGGER.logMessage(message_translated, 'MetaSearch', getattr(Qgis, level.capitalize(), 'Info'))
```

Hi @tomkralidis, I would not use a string (e.g. `Info`, `Warn` etc.) to specify the level, but use dedicated methods (`log_info`, `log_warning` etc.) instead, but maybe that's overkill.
You could also make sure that the `level` has the correct capitalization, and - if the attribute is not found - set a default log level?

Anyway, thanks for adding some logging! It helps to see what MetaSearch is doing under the hood :)

**Review by GeoSander — APPROVED** [review]

File: python/plugins/MetaSearch/util.py (hunk 1 above)
Comment (tomkralidis) [reply python/plugins/MetaSearch/util.py]: Thanks @GeoSander.  I've pushed an update just now that adds a default (`Info`).

**Comment by tomkralidis** [comment]
CI failure is unrelated.

File: python/plugins/MetaSearch/util.py (hunk 1 above)
Comment (nyalldawson) [reply python/plugins/MetaSearch/util.py]: Why not just use Qgis.MessageLevel directly here instead? The current approach with passing a string and then resolving via getattr looks needlessly fragile.

File: python/plugins/MetaSearch/util.py [hunk 2]
```diff
@@ -167,3 +166,10 @@ def clean_ows_url(url):
         url.setQuery(query_string)
 
     return url.toString()
+
+
+def log_message(message, level='Info'):
+    """helper function to emit logging messages"""
+
+    message_translated = QCoreApplication.translate('MetaSearch', message)
```
Comment (nyalldawson) [inline python/plugins/MetaSearch/util.py:174]: This won't actually work for translations, as the strings to translate can't be collected in advance via this approach. You need to translate the strings before calling log_message, and then using them directly here.

File: python/plugins/MetaSearch/util.py (hunk 1 above)
Comment (tomkralidis) [reply python/plugins/MetaSearch/util.py]: Updated.

File: python/plugins/MetaSearch/util.py (hunk 2 above)
Comment (tomkralidis) [reply python/plugins/MetaSearch/util.py]: updated.

File: python/plugins/MetaSearch/search_backend.py [hunk 3]
```diff
@@ -84,6 +89,8 @@ def __init__(self, url, timeout, username, password, auth):
         self.record_info_template = 'record_metadata_dc.html'
         self.constraints = []
 
+        log_message(QCoreApplication.translate(
+            'MetaSearch', f'Connecting to CSW: {self.url}'), Qgis.Info)
```
Comment (Gustry) [inline python/plugins/MetaSearch/search_backend.py:93]: Fstrings doesn't work with translations I think. Same comment as @nyalldawson, the string will be collected in advance, but with fstring, the string will change. I personally keep using `.format`. Am I wrong ?

Side note, I thought we do not translate logs in QGIS, or in general. Logs is for advanced users or developers. Imagine you receive logs in another language, it makes it very difficult to debug.

Comment (tomkralidis) [reply python/plugins/MetaSearch/search_backend.py]: Thanks @Gustry.  Removed translation functionality from log messages.

**Review by nyalldawson — APPROVED** [review]

**Comment by agiudiceandrea** [comment]
Hi @tomkralidis, I would like the logging was optional (off by default), maybe adding a checkbox in the plugin settings dialog window. Is it possible?

**Comment by tomkralidis** [comment]
Associated PR in #60049

**Pull request merged**